### PR TITLE
Add tests & fix issues

### PR DESCRIPTION
**Pull request by Bogdan-D**
# Submit a pull request

## CLA

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required).
- [x] The code changes follow best practices
- [x] Code changes are tested (add some information if not applicable)

### CODE REVIEW

File: channel_type.go [hunk 1]
```diff
@@ -1,30 +1,63 @@
...
-	err = c.makeRequest(http.MethodGet, "channeltypes/"+chanType, nil, nil, &resp)
+func (c *Client) GetChannelType(chanType string) (ChannelType, error) {
+	var resp struct {
+		ChannelTypes map[string]ChannelType `json:"channel_types"`
+	}
 
-	return
+	err := c.makeRequest(http.MethodGet, "channeltypes/"+chanType, nil, nil, &resp)
+
+	return resp.ChannelTypes[chanType], err
```
Comment (pascaldekloe) [inline channel_type.go]: Check presence with error.

File: client.go [hunk 2]
```diff
@@ -98,21 +98,21 @@ func (c *Client) makeRequest(method string, path string, params map[string][]str
...
 	params := map[string]interface{}{
 		"user_id": userID,
 	}
 
-	return c.createToken(params, expire)
+	return c.createToken(params, expire...)
 }
 
-func (c *Client) createToken(params map[string]interface{}, expire *time.Time) ([]byte, error) {
+func (c *Client) createToken(params map[string]interface{}, expire ...time.Time) ([]byte, error) {
```
Comment (pascaldekloe) [inline client.go]: The vararg is too hacky for auth stuff, even when it's a private method. 😬

File: utils_test.go [hunk 3]
```diff
@@ -37,3 +40,9 @@ func mustNoError(t *testing.T, err error) {
 		t.FailNow()
 	}
 }
+
+func mustError(t *testing.T, err error) {
```
Comment (pascaldekloe) [inline utils_test.go]: Looks like an essential feature for testify. I've skipped that argument, due to your time pressure. 😁

File: user.go [hunk 4]
```diff
@@ -72,8 +50,8 @@ func (c *Client) UnmuteUser(targetID string, userID string) error {
 }
 
 func (c *Client) FlagUser(targetID string, options map[string]interface{}) error {
-	if options == nil {
-		return errors.New("flag user: options are nil")
+	if options == nil || len(options) == 0 {
```
Comment (pascaldekloe) [inline user.go]: The len function returns zero for nil maps (and slices).

File: channel_type.go (hunk 1 above)
Comment (Bogdan-D) [reply channel_type.go]: API will return 404 it it's not exists. But let's do not trust API :)

File: utils_test.go (hunk 3 above)
Comment (Bogdan-D) [reply utils_test.go]: :+1:

File: channel.go [hunk 5]
```diff
@@ -1,48 +1,103 @@
...
-	channelPathFmt = "channels/%s/%s"
-	messagePathFmt = "channels/%s/%s/message"
-	eventPathFmt   = "channels/%s/%s/event"
-)
+type ChannelMember struct {
+	UserID      string `json:"user_id,omitempty"`
+	User        *User  `json:"user,omitempty"`
+	IsModerator bool   `json:"is_moderator,omitempty"`
+
+	Invited          bool       `json:"invited,omitempty"`
```
[the remark this answers is not in the document]
Comment (Bogdan-D) [reply channel.go]: But it's returned from API, maybe let's stay close to response body?

[the remark this answers is not in the document]
Comment (pascaldekloe) [reply channel.go]: I'll add it to the beta API TODO list. 👍

**Pull request merged**